### ISSUE TITLE
Expose types and functions for deterministic maths

### ISSUE THREAD

**Issue by TornadoStorm** [appsinacup/godot-rapier-physics#226]
It would be useful to be able to utilize the types and functions used for Rapier's cross-platform deterministic addons (such as its internally used fixed point or soft floating-point types, whichever is used). This would be very useful in writing deterministic code for multiplayer games.

**Comment by Ughuuu** [comment]
Any ideas on how to do this? Expose new classes? Eg RapierVector2? RVector2?
It would be quite a lot of stuff to expose too. What classes and functions you thinking of specifially?

**Comment by TornadoStorm** [comment]
Yes, RapierVector2 implemented as a type would be useful, along with its Vector3 counterpart, along with their corresponding implementations of addition, multiplication, dot product etc..

Or if at least whatever Rapier uses in place of floats could be used in godot as a type this could also make it easier for people to make their own implementations of vectors using this type.

**Comment by Ughuuu** [comment]
Right, I know of Vector2 and Vector3, can you make a list of all the things that would be needed by this? Eg. Transform2D/3D, etc.

**Comment by Ughuuu** [comment]
For start adding the standard ComplexField types (eg. sin, cos, sqrt, etc). These are then used inside vector2, etc. So with these there need to be higher level classes/etc rewritten. Someone could use these to make sure the game is deterministic though (eg instead of computing length of a vector, use ComplexField sqrt to compute it).

**Comment by Ughuuu** [comment]
Also, an interesting read on what you need to take care and what not when going for determinism is this article from Rapier: https://rapier.rs/docs/user_guides/rust/determinism/
